### PR TITLE
[Merged by Bors] - Doc warnings

### LINKED ISSUE #2576: Documentation warnings
## How can Bevy's documentation be improved?

There are various warnings when building documentation for the bevy crates.

Going through the list:
bevy_app:
 - `FromResources` does not exist (nor does it exist in any other crate, I believe). I think this one is from a refactoring to `FromWorld`.
 - `WinitPlugin`, `DefaultPlugins`, and `MinimalPlugins` are all missing because they are in different crates. 

bevy_ecs:
 - broken links for `Bundle::component_id` (type of `component_ids`?) 
 - broken link `SystemParam` (not imported, needs full path).

bevy_pbr and bevy_asset both have bare urls that should be surrounded with <>.
The urls are an easy fix (assuming we want to make them clickable?)

For the refactoring and typos, if that is what those are, that should be easy to fix as well.

The rest of the crates give no errors, so good there.

The different crate plugins are referenced from bevy_app, but are in bevy_internal, which poses an issue.

### PR DESCRIPTION
**Pull request by Hoidigan**
Fixes #2576 

Add <> to links where needed, fix typo and refactored names, and add docs.rs links for external items.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
bors try

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
bors try

**Comment by Hoidigan** [comment]
@cart Could I get the workflow run for this? I can't do it since I'm a first time contributor.

**Comment by alice-i-cecile** [comment]
bors try

**Comment by Hoidigan** [comment]
What should I do to get that to work?
It says to run cargo fmt --all, which modifies uh... 408 files, it looks like.
And whatever the dependencies issue is?

**Comment by alice-i-cecile** [comment]
@Hoidigan, clicking through, I'm seeing

> error: failed to download from `https://crates.io/api/v1/crates/ab_glyph_rasterizer/0.1.4/download`

> Caused by:
>  [55] Failed sending data to the peer (Connection died, tried 5 times before giving up)
> Error: Process completed with exit code 101.

Looks like a networking error that isn't your fault. https://github.com/bevyengine/bevy/pull/2040 has more contributing advice for you though; it's just waiting on final review now <3

**Comment by alice-i-cecile** [comment]
bors try

**Review by NiklasEi — APPROVED** [review]

File: crates/bevy_app/src/app.rs
Comment (mockersf) [inline crates/bevy_app/src/app.rs:444]: full links to docs.rs should be avoided

File: crates/bevy_pbr/src/material.rs
Comment (mockersf) [inline crates/bevy_pbr/src/material.rs:53]: this is not a doc comment but just a comment, `<..>` should not be needed here

File: crates/bevy_pbr/src/material.rs
Comment (mockersf) [inline crates/bevy_pbr/src/material.rs:63]: not a doc comment but a comment

File: crates/bevy_app/src/app.rs
Comment (Hoidigan) [reply crates/bevy_app/src/app.rs:444]: What should I change it to get the correct link? Replace the 0.5.0 with *?
We can't point it to WinitPlugin from inside that crate, because it doesn't have that as a dependency.

Comment (mockersf) [reply crates/bevy_app/src/app.rs:444]: you should be able to add a dev dependencies to `bevy_winit` that will allow doc links to work 😃

Comment (Hoidigan) [reply crates/bevy_app/src/app.rs:444]: I don't think it does... It doesn't bring it into scope for the doc comments. I'm fairly certain we'd have to bring it in as a full dependency.
https://github.com/rust-lang/api-guidelines/discussions/186

Oh here we are! I was trying to dig and find this! https://discord.com/channels/691052431525675048/743559241461399582/871090893162156082

Comment (jyn514) [reply crates/bevy_app/src/app.rs:444]: Cargo doesn't provide dev-dependencies when documenting. @nemo157 has suggested `doc-dependencies` a few times but it doesn't currently exist.

See also https://github.com/rust-lang/rust/issues/74481.

Comment (mockersf) [reply crates/bevy_app/src/app.rs:444]: Oh right, dev dependencies are available when testing doc code blocks but not when building docs...

The other place where we have a similar link is https://github.com/bevyengine/bevy/blob/90586a4c461cde86b8a0e08f0770894e7cf30862/crates/bevy_ecs/src/event.rs#L120

If we consider than the main target for those links will be docs.rs users, I think it should link to the root crate (`bevy` instead of `bevy_winit`) and use `*` for version. What do you think?

Comment (Hoidigan) [reply crates/bevy_app/src/app.rs:444]: I think that sounds good, I'll go through and do that.

**Comment by mockersf** [comment]
@Hoidigan could you run `cargo fmt` to fix format?

**Comment by Hoidigan** [comment]
I should be able to this Sunday, sorry about the wait. I'm away and don't have my computer

**Comment by Hoidigan** [comment]
(Sorry I'm a bit slow)

Running cargo fmt --all modifies all 408 files. I only committed the changes in the 3 ci complained about, but that was little weird. I think it was newline issues, but I don't actually know.

**Review by mockersf — APPROVED** [review]

**Pull request closed without merging**
